### PR TITLE
Use Ecto.embedded_load/3

### PR DESCRIPTION
**Pull request by maennchen**

### CODE REVIEW

File: lib/polymorphic_embed.ex [hunk 1]
```diff
@@ -118,26 +118,19 @@ defmodule PolymorphicEmbed do
...
   @impl true
-  def dump(%_module{} = struct, _dumper, %{metadata: metadata}) do
-    Ecto.Type.dump(:map, map_from_struct(struct, metadata))
+  def dump(%_module{} = struct, dumper, %{metadata: metadata}) do
+    dumper.(:map, map_from_struct(struct, metadata))
   end
 
-  def dump(nil, _dumper, _params) do
-    Ecto.Type.dump(:map, nil)
+  def dump(nil, dumper, _options) do
```
Comment (mathieuprog) [inline lib/polymorphic_embed.ex:132]: in the signature it is written params, not options:
https://hexdocs.pm/ecto/Ecto.ParameterizedType.html#c:dump/3

#OCD

**Review by mathieuprog — APPROVED** [review]

File: lib/polymorphic_embed.ex (hunk 1 above)
Comment (maennchen) [reply lib/polymorphic_embed.ex]: @mathieuprog Changed :)

**Comment by mathieuprog** [comment]
And another great contribution:)

**Comment by mathieuprog** [comment]
By the way you simplified dump and load, and do you think it's possible to simplify cast as well then?
(there are some functionalities like allowing to cast an embed without the need for a changeset function, but we can remove all of that if the code can be more simple like the dump and load refactoring)

**Comment by maennchen** [comment]
@mathieuprog I have an idea how that could work, let me try.

**Comment by mathieuprog** [comment]
Again, if some tests break it is expected. Maybe there will be some functionality that I added that a normal embed wouldn't support, not sure. I prefer simpler code and behavior similar to normal embeds.

**Comment by maennchen** [comment]
@mathieuprog See #26

**Pull request merged**